jupyter_server/services/sessions/handlers.py

```diff
@@ -161,14 +161,30 @@ async def patch(self, session_id):
                 changes["kernel_id"] = kernel_id
             elif model["kernel"].get("name") is not None:
                 kernel_name = model["kernel"]["name"]
-                kernel_id = await sm.start_kernel_for_session(
-                    session_id,
-                    kernel_name=kernel_name,
-                    name=before["name"],
-                    path=before["path"],
-                    type=before["type"],
-                )
-                changes["kernel_id"] = kernel_id
+
+                try:
+                    kernel_id = await sm.start_kernel_for_session(
+                        session_id,
+                        kernel_name=kernel_name,
+                        name=before["name"],
+                        path=before["path"],
+                        type=before["type"],
+                    )
+                    changes["kernel_id"] = kernel_id
+                except Exception as e:
+                    # the error message may contain sensitive information, so we want to
+                    # be careful with it, thus we only give the short repr of the exception
+                    # and the full traceback.
+                    # this should be fine as we are exposing here the same info as when we strt a new kernel
+                    msg = "The '%s' kernel could not be started: %s" % (
+                        kernel_name,
+                        repr(str(e)),
+                    )
+                    status_msg = "Error starting kernel %s" % kernel_name
+                    self.log.error("Error starting kernel: %s", kernel_name)
```

Comment:
```suggestion
                    self.log.error(status_msg)
```

Reply:
No; you don't use formatted strings in loggers because of structured logging; which is exactly why I added rule G001

Reply:
Note that I did not enable G002, which also prevent g-strings in logs and there are plenty of those, including a fews where i18n is wrong and will never match as the formatting is done before i18n.

```
jupyter_server/auth/identity.py:                self.log.info(_i18n(f"Wrote hashed password to {config_file}"))
jupyter_server/serverapp.py:        self.log.info(_i18n("Writing Jupyter server cookie secret to %s"), self.cookie_secret_file)
```

Reply:
> No; you don't use formatted strings in loggers because of structured logging; which is exactly why I added rule G001

If the reason is to not compute the string ahead of time, you already did it in line 183.

Reply:
> g-strings

Hopefully we don't have those in logs :smile:

Reply:
> If the reason is to not compute the string ahead of time, you already did it in line 183.

No structured logging, if you have a Handler that forward to a logging storage system, the log db will index by the first argument ("Error starting kernel: %s") and allow things like "show all similar log messages", this avoid having to use regular expressions, and when the arguments are numerics, many logs system  can automatically give you stat on those numbers. 

Arguably here it's almost inconsequential; but as a matter of good practice (and to give nice training set to AI 😏 ), it's better to do it everywhere.